test/test_jit_cuda_fuser.py

```diff
@@ -4152,6 +4152,50 @@ def t(x):
         t_jit = torch.jit.script(t)
         self._run_helper(t_jit, t, x)
 
+    @unittest.skipIf(ALIAS_TEST_DISABLED, "skipping this test since index_select's backend PR hasn't been merged")
+    @unittest.skipIf(not RUN_NVFUSER, "requires CUDA")
+    @unittest.skipIf(GRAPH_EXECUTOR != ProfilingMode.PROFILING,
+                     "Requires fusion optimization pass to be effective")
+    def test_index_select_fusion(self):
+        lookup_size = 68
+        feat_dim = 128
+        num_elements = 355984
+        lookup_tv = torch.rand(lookup_size, feat_dim, dtype=torch.float, device="cuda")
+        indies_tv = torch.randint(0, lookup_size, (num_elements,), device="cuda").to(dtype=torch.int)
+        sbf = torch.rand(num_elements, feat_dim, dtype=torch.float, device="cuda")
+
+        def t(x_kj, idx_kj, sbf):
+            sbf_res = torch.index_select(x_kj, 0, idx_kj) * sbf
+            sbf_res = sbf_res + 17
+            return sbf_res
+        t_jit = torch.jit.script(t)
+        self._run_helper(t_jit, t, lookup_tv, indies_tv, sbf)
+
+    @unittest.skipIf(ALIAS_TEST_DISABLED, "skipping this test since index_select's backend PR hasn't been merged")
+    @unittest.skipIf(not RUN_NVFUSER, "requires CUDA")
+    @unittest.skipIf(GRAPH_EXECUTOR != ProfilingMode.PROFILING,
+                     "Requires fusion optimization pass to be effective")
+    def test_index_select_runtime_dim(self):
+        lookup_size = 68
+        feat_dim = 128
+        num_elements = 355984
+        dim = torch.tensor(0, device='cuda').to(dtype=torch.int)
+        lookup_tv = torch.rand(lookup_size, feat_dim, dtype=torch.float, device="cuda")
+        indies_tv = torch.randint(0, lookup_size, (num_elements,), dtype=torch.float, device="cuda").to(dtype=torch.int)
+        sbf = torch.rand(num_elements, feat_dim, dtype=torch.float, device="cuda")
+
+        def failure(x_kj, idx_kj, sbf):
```

Comment:
We should be able to support runtime scalar input. e.g. https://github.com/pytorch/pytorch/blob/391b593ca262432ccba1939f7448275cfd4f62e6/test/test_jit_cuda_fuser.py#L1106-L1110 

runtime_dim shouldn't be a failure if the scalar is passed properly.
```
def func(x_kj, idx_kj, sbf, dim: int):
    sbf_res = torch.index_select(x_kj, dim, idx_kj) * sbf
    sbf_res = sbf_res + 17
    return sbf_res
```

And you would also need to run it like `self._run_helper(t_jit, t, lookup_tv, indies_tv, sbf, 0)`